tests/playbooks/roles/install-csi-cinder/tasks/main.yaml

```diff
@@ -0,0 +1,215 @@
+                  - /bin/openstack-cloud-controller-manager
+                  - --v=4
+                  - --cloud-config=/etc/config/cloud.conf
+                  - --cloud-provider=openstack
+                  - --use-service-account-credentials=true
+                  - --bind-address=127.0.0.1
+                volumeMounts:
+                  - mountPath: /etc/config
+                    name: cloud-config-volume
+                    readOnly: true
+            hostNetwork: true
+            volumes:
+            - name: cloud-config-volume
+              secret:
+                secretName: cloud-config
+      EOF
+
+- name: Wait for openstack-cloud-controller-manager up and running
+  shell:
+    executable: /bin/bash
+    cmd: |
+      kubectl -n kube-system get pod | grep openstack-cloud-controller-manager | grep Running
+  register: check_occm
+  until: check_occm.rc == 0
+  retries: 24
+  delay: 5
+  ignore_errors: yes
+
+- name: Deploy cinder-csi-plugin
+  shell:
+    executable: /bin/bash
+    cmd: |
+      cd $GOPATH/src/k8s.io/cloud-provider-openstack
+      kubectl apply -f manifests/cinder-csi-plugin
+  ignore_errors: yes
+
+- name: Wait for csi-cinder-controllerplugin up and running
+  shell:
+    executable: /bin/bash
+    cmd: |
+      kubectl -n kube-system get pod | grep csi-cinder-controllerplugin  | grep Running
+  register: check_csi_controller
+  until: check_csi_controller.rc == 0
+  retries: 24
+  delay: 5
+  ignore_errors: yes
+
+- name: Wait for csi-cinder-nodeplugin up and running
+  shell:
+    executable: /bin/bash
+    cmd: |
+      kubectl -n kube-system get pod | grep csi-cinder-nodeplugin  | grep Running
+  register: check_csi_node
+  until: check_csi_node.rc == 0
+  retries: 24
+  delay: 5
+  ignore_errors: yes
+
+- name: Gather additional evidence if csi-cinder-plugin failed to come up
+  when: check_csi_controller.failed or check_csi_node.failed
+  block:
+    - name: Describe failed csi-cinder-plugin
+      shell:
+        executable: /bin/bash
+        cmd: |
+          kubectl get pods -A
+          kubectl -n kube-system describe deployment csi-cinder-controllerplugin
+          kubectl -n kube-system describe daemonset csi-cinder-nodeplugin
+      register: describe_csi
+      changed_when: false
+
+    - name: Log failed csi-cinder-plugin deployment
+      debug:
+        var: describe_csi.stdout_lines
+
+    - name: &failmsg Stop due to prior failure of csi-cinder-plugin
+      fail:
+        msg: *failmsg
+
+- name: Run functional tests for csi-cinder-plugin
+  shell:
+    executable: /bin/bash
+    cmd: |
+      set -x
+      set -e
+      set -o pipefail
+
+      cd $GOPATH/src/k8s.io/cloud-provider-openstack
+
+      ARTIFACTS="${ARTIFACTS:-${PWD}/_artifacts}" go test -v ./cmd/tests/cinder_csi_e2e_suite_test.go -ginkgo.v -ginkgo.progress -ginkgo.skip="\[Disruptive\]" -ginkgo.focus="\[cinder-csi-e2e\]" -ginkgo.noColor -timeout=0 -report-dir='{{ k8s_log_dir }}' | tee "{{ k8s_log_dir }}/e2e.log"
```

Comment:
@jichenjc where can i check e2e.log as mentioned here? I can only open build-log.txt

Reply:
I made a change and will check result later on , you are right, I can't find the log as well

Reply:
the PR folder is 
pr-logs/pull/cloud-provider-openstack/1640/openstack-cloud-csi-cinder-e2e-test/1441935266414596096

I don't know which variable can be used ,anyway, as build.log already have all logs,
I think we can avoid the tee for now, later on ,we need put all logs including pod logs etc into some folder